src/components/JsonSchemaViewer.tsx

```diff
@@ -44,8 +57,9 @@ const JsonSchemaViewerComponent: React.FC<JsonSchemaProps & ErrorBoundaryForward
       return !((viewMode === 'read' && !!validations.writeOnly) || (viewMode === 'write' && !!validations.readOnly));
     });
     jsonSchemaTree.populate();
+    onTreePopulated?.(jsonSchemaTree);
```

Comment:
I know that this `useMemo` is quite side-effect'y already, but we really shouldn't do that.

Simple refactor would allow for that function to be called in a separate `useEffect` hook.

Reply:
Can you elaborate more on what kind of refactor you have in mind?

Reply:
tracking `isPopulated` state and then using that in `useEffect`?

Reply:
oh I guess I could just use `jsonSchemaTreeRoot` as dependency, I will try it out

Reply:
`useMemo` can return the whole jsonSchemaTree.

Than you just make `useEffect` where that callback is called with `jsonSchemaTree` 🤷

Reply:
something like:

```js
const jsonSchemaTree = React.useMemo(...);

React.useEffect(() => {
   onTreePopulated?.(jsonSchemaTree);
}, [jsonSchemaTree, onTreePopulated]);
```